openstudiocore/src/utilities/idf/Test/IdfObject_GTest.cpp

```diff
@@ -743,3 +742,72 @@ TEST_F(IdfFixture, DoubleDisplayedAsString) {
   EXPECT_DOUBLE_EQ(value,roundTripValue);
 }
 
+TEST_F(IdfFixture, IdfObject_SetDouble_NaN_and_Inf) {
+
+  // try with an IdfObject
+  // IdfObject does not prevent Infinity and NaN
+  IdfObject object(IddObjectType::OS_People_Definition);
+
+  // Set Number of People
+  // Check for nan
+  EXPECT_TRUE(object.setDouble(3, std::numeric_limits<double>::quiet_NaN()));
```

Comment:
@jmarrec check out the difference between EXPECT_ and ASSERT_ in gtest.  EXPECT statements will fail but allow the test to continue, ASSERTS immediately exit on first failure.  It's helpful to see multiple failures at once if possible

Reply:
I understand the difference between the two, it's a bit of a philosophical question here: I consider that if any of these fails, I must fix it. My "internal" policy was to always use ASSERT (coming from Python, that's what I'm used to anyways), but if EXPECT are preferred I'm fine with it, I just didn't know if was a thing.

I'm pretty sure it's quite inconsistent in the existing GTests though. And you should expect (no pun intended) the same behavior on all the Gtests I've written so far unfortunately :/

If you want to be able to see as much messages as possible, perhaps you could make it a policy to only use EXPECT and no ASSERT (it's easier to remember and apply (because it's binary) than saying "prefer EXPECT")

I understand that ASSERT is for stuff where it's meaningless to continue if it fails, but if you take the following example, I don't think it provides more valuable input:

On line 758 to 760 below, you wrote:
```
EXPECT_TRUE(object.setDouble(3, std::numeric_limits<double>::infinity()));
ASSERT_TRUE(object.getDouble(3));
EXPECT_TRUE(std::isinf(object.getDouble(3).get()));
```
why use an ASSERT here? It'll crash on the next line if the it wasn't ok anyways, so that isn't going to provide more valuable input, is it?

Reply:
Crashes and seg faults don't provide as nice of output, so my standard is to use an ASSERT to prevent a crash or seg fault, but EXPECT's otherwise.  It is a matter a preference, i just like to see more output from a failed test if possible